### PR TITLE
RSDK-4458: Implement wheeled encoder reconfigure

### PR DESCRIPTION
**Pull request by martha-johnston**
This PR implements the `Reconfigure` function for the wheeled odometry model, as well as increasing the time interval threshold per RSDK-4599

### CODE REVIEW

File: components/movementsensor/wheeledodometry/wheeledodometry_test.go [hunk 1]
```diff
@@ -110,6 +116,51 @@ func TestNewWheeledOdometry(t *testing.T) {
...
+			Base:              newBaseName,
+			TimeIntervalMSecs: 300,
+		},
+	}
+
+	err = fakeSensor.Reconfigure(ctx, newDeps, newconf)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, od.timeIntervalMSecs, test.ShouldEqual, 300)
+	test.That(t, od.baseWidth, test.ShouldEqual, 0.2)
+	test.That(t, od.wheelCircumference, test.ShouldEqual, 0.2)
```
Comment (oliviamiller) [inline components/movementsensor/wheeledodometry/wheeledodometry_test.go:161]: optional but might want to add another test case where not everything in the config changes and testing that the attributes that are supposed to stay the same don't change. but if you've also done manual testing for this then prob fine!

**Review by oliviamiller — CHANGES_REQUESTED** [review]
just had a few nits/questions! overall might be helpful to add some comments throughout to explain whats happening since its not the most intuitive reconfigure.

File: components/movementsensor/wheeledodometry/wheeledodometry.go [hunk 2]
```diff
@@ -102,85 +102,138 @@ func (cfg *Config) Validate(path string) ([]string, error) {
...
-// newWheeledOdometry returns a new wheeled encoder movement sensor defined by the given config.
-func newWheeledOdometry(
-	ctx context.Context,
-	deps resource.Dependencies,
-	conf resource.Config,
-	logger golog.Logger,
-) (movementsensor.MovementSensor, error) {
+// Reconfigure automatically reconfigures this movement sensor based on the updated config.
+func (o *odometry) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	if len(o.motors) > 0 {
```
[the remark this answers is not in the document]
Comment (martha-johnston) [reply components/movementsensor/wheeledodometry/wheeledodometry.go:107]: right now we are only supporting 1 set of motors so there's a couple things that will have to change when we start supporting more

File: components/movementsensor/wheeledodometry/wheeledodometry_test.go (hunk 1 above)
Comment (martha-johnston) [reply components/movementsensor/wheeledodometry/wheeledodometry_test.go:161]: good point, will do!

File: components/movementsensor/wheeledodometry/wheeledodometry.go [hunk 3]
```diff
@@ -102,85 +102,138 @@ func (cfg *Config) Validate(path string) ([]string, error) {
...
-	for i := range newConf.LeftMotors {
-		motorLeft, err := motor.FromDependencies(deps, newConf.LeftMotors[i])
-		if err != nil {
-			return nil, err
-		}
-		properties, err := motorLeft.Properties(ctx, nil)
-		if err != nil {
-			return nil, err
+	if o.motors != nil && len(o.motors) == len(newConf.LeftMotors) {
+		isSame := true
```
[the remark this answers is not in the document]
Comment (martha-johnston) [reply components/movementsensor/wheeledodometry/wheeledodometry.go:152]: actually realizing I don't think this check is necessary. going to remove it and test

**Review by oliviamiller — APPROVED** [review]

**Pull request merged**